From 939bd5baadb18f56235c3bf9a2fa08e30e8d336a Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Wed, 20 Aug 2014 04:39:04 +0200
Subject: [PATCH 4/9] sparc testsuite compile fix

tst-spin1.c compile breaks with:
test/nptl/tst-spin1.c:34: undefined reference to `pthread_spin_lock'

pthread_spin_lock and pthread_spin_trylock is missing while
building sparc. add the meta c files here.

Signed-off-by: Waldemar Brodkorb <wbx@uclibc-ng.org>
---
 libpthread/nptl/sysdeps/sparc/pthread_spin_lock.c    | 5 +++++
 libpthread/nptl/sysdeps/sparc/pthread_spin_trylock.c | 5 +++++
 2 files changed, 10 insertions(+)
 create mode 100644 libpthread/nptl/sysdeps/sparc/pthread_spin_lock.c
 create mode 100644 libpthread/nptl/sysdeps/sparc/pthread_spin_trylock.c

diff --git a/libpthread/nptl/sysdeps/sparc/pthread_spin_lock.c b/libpthread/nptl/sysdeps/sparc/pthread_spin_lock.c
new file mode 100644
index 0000000..dcc5ae2
--- /dev/null
+++ b/libpthread/nptl/sysdeps/sparc/pthread_spin_lock.c
@@ -0,0 +1,5 @@
+#if defined(__arch64__)
+#include "sparc64/pthread_spin_lock.c"
+#else
+#include "sparc32/pthread_spin_lock.c"
+#endif
diff --git a/libpthread/nptl/sysdeps/sparc/pthread_spin_trylock.c b/libpthread/nptl/sysdeps/sparc/pthread_spin_trylock.c
new file mode 100644
index 0000000..af63eec
--- /dev/null
+++ b/libpthread/nptl/sysdeps/sparc/pthread_spin_trylock.c
@@ -0,0 +1,5 @@
+#if defined(__arch64__)
+#include "sparc64/pthread_spin_trylock.c"
+#else
+#include "sparc32/pthread_spin_trylock.c"
+#endif
-- 
1.8.5.2 (Apple Git-48)

